spectractor/simulation/simulator.py

```diff
@@ -431,11 +430,14 @@ def simulate(self, A1=1.0, A2=0., aerosols=0.05, ozone=300, pwv=5, D=parameters.
         self.my_logger.debug(f'\n\tAfter build ima1: {time.time() - start}')
 
         # Add order 2
-        if A2 > 0.:
+        if A2 > 0. and self.disperser.ratio_order_2over1 is not None:
             spectrum_order2 = self.disperser.ratio_order_2over1(self.lambdas) * spectrum
             spectrum_order2_err = self.disperser.ratio_order_2over1(self.lambdas) * spectrum_err
             if np.any(np.isnan(spectrum_order2)):
                 spectrum_order2[np.isnan(spectrum_order2)] = 0.
+            dispersion_law_order2 = self.compute_dispersion_in_spectrogram(self.lambdas, shift_x, shift_y, angle,
+                                                                           niter=5, with_adr=True,
+                                                                           order=self.order + np.sign(self.order))
```

Comment:
I'm not quite sure what the  `np.sign(self.order)` part is going, but adding it feels strange, as opposed to multiplying by it, given that it's +/-1. Are you sure that's right? (There's a similar line somewhere above too)

Reply:
Yes I am sure: if specified order is negative, then next order is order-1, and second next order is order-2, so you take the sign of the given order (np.sign returns -1)  and add it.

Reply:
Haha, OK, that makes sense, thanks 🙂